### PR TITLE
add hashery library first pass

### PR DESCRIPTION
**Pull request by tongxiang**
#### What's this PR do?

This PR translates [Hashery.java](https://github.com/DoSomething/hashery.android) into Objective-C for the DoSomething iOS app. It works almost exactly the same way. 
#### Where should the reviewer start?

`DSOHashery.m`. 
#### How should this be manually tested?

Right now, I've left some noob-y code (to be removed later) in main.m which tests the functionality of the app. Feel free to play around with those functions in order to confirm tests. Left in some `NSLog` statements to make it clear what functions are doing--to be removed later. 

Creates an array of arrays. The number of arrays within the array becomes the number of words in the hash string. Each array needs to have the same number of words. 

```
        NSArray *wordArray = @[
            @[@"big", @"tall", @"short", @"husky"],
            @[@"orange", @"grey", @"purple", @"vermilion"],
            @[@"monkey", @"sloth", @"giraffe", @"dolphin"]
        ];
```

Instantiates a hashery object. 

```
        DSOHashery *hashery = [[DSOHashery alloc] init];
```

Sets the hashery object to use the wordArray. 

```
        [hashery setWordArrayLists:wordArray];
```

Encodes `13` into a unique string. 

```
        [hashery encodeBase10ToUniqueString:13];
```

Decodes a unique string into an integer. 

```
        [hashery decodeUniqueStringToBase10:@"bigvermilionsloth"];
```
#### Any background context you want to provide?

Todo next: convert this to a cocoapod. 
#### Questions:

Would _love_ all suggestions for improvement.

### CODE REVIEW

File: hashery/DSOHashery.m [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    
+    _base = [[_words objectAtIndex:0] count];
+    
+    NSInteger base = _base;
+    for (int i = 1; i < _words.count; i++) {
+        base *= _base;
+    }
+    _maxIntegerValueOfHash = base - 1;
+    NSLog(@"_maxIntegerValueOfHash: %zd", _maxIntegerValueOfHash);
+    return true;
```
Comment (aaronschachter) [inline hashery/DSOHashery.m]: Best practice is to use `YES` and `NO` for Booleans: https://github.com/DoSomething/ios-RBReviewer/issues/47

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Hashery
+
+This is an Objective-C library that converts numbers into human-readable word combinations.
+
+#### Get the library
+
+- Download the latest release: https://github.com/DoSomething/hashery.android/releases
+- Add the `hashery.jar` file to your project's libs folder
```
Comment (aaronschachter) [inline README.md]: Noooo

Comment (aaronschachter) [inline README.md]: I think all these docs should change to be accurate, or just remove for now

**Comment by aaronschachter** [comment]
Code looks good except for little things noted.

One suggestion is that it might make sense to use a singleton class for the `DSOHashery` class, http://www.galloway.me.uk/tutorials/singleton-classes/, so you load the Hashery object with its data one time and can reference it from other classes easily.

Let me know if you need any help with the cocoapod stuff!

**Comment by tongxiang** [comment]
Thanks for these suggestions, @aaronschachter! Will definitely be working on converting this to a cocoapod later this week, would be great to get your advice when I get to that point.

**Pull request merged**